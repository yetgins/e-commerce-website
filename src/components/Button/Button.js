import React from "react";
import styled from "styled-components";

const Button = ({ text, onClick, height, width,background }) => {
     const ButtonWrapper = styled.button`
    background: ${background? background :'#f0c14b' };
    border-radius: 2px;
    border: 1px solid;
    margin-top: 10px;
    border-color: #a88734 #9c7e31 #846a29;
    color: #111;
    height: ${height};
    width: ${width};
  `;
  return (
    <div>
      <ButtonWrapper className="button" data-testid="button" onClick={onClick}>
        {text}
      </ButtonWrapper>
    </div>
  );
};

export default Button;
