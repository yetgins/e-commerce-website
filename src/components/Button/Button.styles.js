import styled from 'styled-components';

export const ButtonWrapper=styled.button`
    background: ${({background})=> background? background :'#f0c14b' };
    border-radius: 2px;
    border: 1px solid;
    margin-top: 10px;
    border-color: #a88734 #9c7e31 #846a29;
    color: #111;
    height: ${({height})=>height? height : ''};
    width: ${({width})=> width? width : ''};
`;