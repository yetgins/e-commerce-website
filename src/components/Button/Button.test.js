import {
  render,
  cleanup,
  fireEvent,
  getByTestId,
} from "@testing-library/react";
import React from "react";
import Button from "./Button";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  render(<Button></Button>, div);
});

it("renders text1 correctly", () => {
  const { getByTestId } = render(<Button text="click me please"></Button>);

  expect(getByTestId("button")).toHaveTextContent("click me please");
});

it("renders text2 correctly", () => {
  const { getByTestId } = render(<Button text="save"></Button>);

  expect(getByTestId("button")).toHaveTextContent("save");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button text="save"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("calls height prop", () => {
  const { getByTestId } = render(<Button height="50px"></Button>);

  expect(getByTestId("button")).toHaveStyle("height: 50px");
});

it("calls width prop", () => {
  const { getByTestId } = render(<Button width="90%"></Button>);

  expect(getByTestId("button")).toHaveStyle("width: 90%");
});

it("calls background prop", () => {
  const { getByTestId } = render(<Button background="white"></Button>);

  expect(getByTestId("button")).toHaveStyle("background: white");
});

it("call background prop", () => {
  const { getByTestId } = render(<Button background=""></Button>);

  expect(getByTestId("button")).toHaveStyle("background: #f0c14b");
});

const onClick = jest.fn();
it("calls onClick prop on button click", () => {
  const { getByTestId } = render(<Button onClick={onClick}></Button>);

  fireEvent.click(getByTestId("button"));
  expect(onClick).toHaveBeenCalled();
});
