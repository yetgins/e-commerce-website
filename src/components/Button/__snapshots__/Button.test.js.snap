// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className="sc-hKgILt lnPSxw button"
    data-testid="button"
  >
    save
  </button>
</div>
`;
